internal/backends/collection.go

```diff
@@ -224,6 +226,42 @@ func (cc *collectionContract) Stats(ctx context.Context, params *CollectionStats
 	return res, err
 }
 
+// ListIndexesParams represents the parameters of Database.ListIndexes method.
+type ListIndexesParams struct{}
+
+// ListIndexesResult represents the results of Database.ListIndexesResult method.
+type ListIndexesResult struct {
+	Indexes []IndexInfo
+}
+
+// IndexInfo represents information about a single index.
+type IndexInfo struct {
+	Unique *bool
+	Name   string
+	Key    IndexKey
+}
+
+// IndexKey is a list of "field name + sort order" pairs.
+type IndexKey []IndexKeyPair
+
+// IndexKeyPair consists of a field name and a sort order that are part of the index.
+type IndexKeyPair struct {
+	Field string
+	Order types.SortType
+}
+
+// ListIndexes returns information about indexes in the database.
+//
+// Database or collection may not exist; that's not an error.
```

Comment:
I think non existent database and collection return error for indexes :thinking:

Reply:
Yes, you are right! Somehow I lost that moment. I will add a test (and fix the logic).